integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala

```diff
@@ -702,10 +716,25 @@ class MergeTestCase extends QueryTest with BeforeAndAfterAll {
       insertExpr(insertMap).
       whenMatched("B.deleted=true").
       delete().execute()
+    assert(getDeleteDeltaFileCount("target") == 1)
     checkAnswer(sql("select count(*) from target"), Seq(Row(3)))
     checkAnswer(sql("select * from target order by key"), Seq(Row("c", "200"), Row("d", "3"), Row("e", "100")))
   }
 
+  private def getDeleteDeltaFileCount(tableName: String): Int = {
+    val table = CarbonEnv.getCarbonTable(None, tableName)(sqlContext.sparkSession)
+    val path = table.getTablePath
```

Comment:
take the path till segment dir, that is till `Part0`

Reply:
I scan the dir with recursive = true. It will get # of all deltafiles.

Reply:
yeah, saw that, but if we give the path till Part directory, we can reduce the files in list, so better to change the path.

Reply:
modified

Reply:
Agree with you. Thanks a lot for this good find.